mungegithub/mungers/sig-mention-handler.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mungers
+
+import (
+	"github.com/golang/glog"
+	githubapi "github.com/google/go-github/github"
+	"github.com/spf13/cobra"
+	"k8s.io/test-infra/mungegithub/features"
+	github "k8s.io/test-infra/mungegithub/github"
+	"reflect"
+)
+
+var (
+	closed          = "closed"
+	currentVersion  = "v1.7"
+	noSigMentionMsg = "There are no @kubernetes/sig mention on the issue. Please mention one."
+	sigLabel        = "sig/"
+)
+
+type SigMentionHandler struct {
+	features *features.Features
+}
+
+func init() {
+	h := &SigMentionHandler{}
+	RegisterMungerOrDie(h)
+}
+
+// Name is the name usable in --pr-mungers
+func (*SigMentionHandler) Name() string { return "sig-mention-handler" }
+
+// RequiredFeatures is a slice of 'features' that must be provided
+func (*SigMentionHandler) RequiredFeatures() []string {
+	return []string{features.RepoFeatureName, features.AliasesFeature}
+}
+
+// Initialize will initialize the munger
+func (s *SigMentionHandler) Initialize(config *github.Config, features *features.Features) error {
+	s.features = features
+	return nil
+}
+
+// EachLoop is called at the start of every munge loop
+func (*SigMentionHandler) EachLoop() error { return nil }
+
+// AddFlags will add any request flags to the cobra `cmd`
+func (*SigMentionHandler) AddFlags(cmd *cobra.Command, config *github.Config) {}
+
+// The algorithm:
+// (1) return false if it is not an issue
+// (2) return false if the issue is closed
+// (3) return false if the issue does not have a v1.7 milestone
+// (4) return false if the issue has any label starting with "sig/"
+// (5) return true otherwise
+func (*SigMentionHandler) IsSigLabelNeeded(obj *github.MungeObject) bool {
+	if obj.Issue == nil || obj.IsPR() || obj.Issue.State == nil || *obj.Issue.State == closed {
+		return false
+	}
+
+	milestone := &githubapi.Milestone{Title: githubapi.String(currentVersion)}
+
+	if obj.Issue != nil && !reflect.DeepEqual(obj.Issue.Milestone, milestone) {
+		return false
+	}
+
+	labels := obj.Issue.Labels
+
+	for i := range labels {
+		label := &labels[i]
```

Comment:
good point! i was writing `label := &labels[i]`, and then `(*(label).Name)`, which was more complicated than needed.